src/webportal/src/app/user/user-view/user-view.component.js

```diff
@@ -250,6 +258,174 @@ const updateUserGithubPAT = (username) => {
+        if (githubPAT) {
+          let req = $.ajax({
+            url: `${webportalConfig.restServerUri}/api/v1/user/${username}/githubPAT`,
+            data: {
+              githubPAT: githubPAT,
+            },
+            type: 'PUT',
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+            dataType: 'json',
+          });
+          reqs.push(req);
+        }
+        // Admin user VC update will be executed in rest-server
+        if (!toBool(admin) && vc) {
+          let req = $.ajax({
+            url: `${webportalConfig.restServerUri}/api/v1/user/${username}/virtualClusters`,
+            data: {
+              virtualClusters: vc,
+            },
+            type: 'PUT',
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+            dataType: 'json',
+          });
+          reqs.push(req);
+        }
+        $.when(...reqs).then(
+          () => {
+            deferredObject.resolve(`User ${username} created successfully`);
+          },
+          (xhr) => {
+            const res = JSON.parse(xhr.responseText);
+            deferredObject.resolve(`User ${username} created successfully but failed when update other info: ${res.message}`);
+          }
+        );
+      },
+      error: (xhr, textStatus, error) => {
+        const res = JSON.parse(xhr.responseText);
+        deferredObject.resolve(`User ${username} created failed: ${res.message}`);
+      },
+    });
+  });
+  return deferredObject.promise();
+};
+
+const addUserRecursively = (userInfos, index, results) => {
+  if (index == 0) {
+    loading.showLoading();
+  }
+  if (index >= userInfos.length) {
+    loading.hideLoading();
+    alert(results.join('\r\n'));
+    window.location.reload();
+  } else {
+    let userInfo = userInfos[index];
+    addUser(userInfo[columnUsername],
+      userInfo[columnPassword],
+      userInfo[columnAdmin],
+      userInfo[columnVC],
+      userInfo[columnGithubPAT])
+      .then((result) => {
+        results.push(result);
+        addUserRecursively(userInfos, ++index, results);
+      });
+  }
+};
+
+const checkUserInfoCSVFormat = (csvResult) => {
+  let fields = csvResult.meta.fields;
+  if (fields.indexOf(columnUsername) === -1) {
+    alert('Missing column of username in the CSV file!');
+    return false;
+  }
+  if (fields.indexOf(columnPassword) === -1) {
+    alert('Missing column of password in the CSV file!');
+    return false;
+  }
+  if (csvResult.errors.length > 0) {
+    alert(`Row ${csvResult.errors[0].row + 2}: ${csvResult.errors[0].message}`);
+    return false;
+  }
+  if (csvResult.data.length == 0) {
+    alert('Empty CSV file');
+    return false;
+  }
+  return true;
+};
+
+const importFromCSV = () => {
+  let readFile = function(e) {
+    let file = e.target.files[0];
+    if (!file) {
+      return;
+    }
+    let reader = new FileReader();
+    reader.onload = function(e) {
+      let contents = e.target.result;
```

Comment:
check `contents` it not null?

Reply:
Yes, it is better to check the contents. Fixed.